gossip-protocol-spec.md

```diff
@@ -102,15 +109,37 @@ struct SomeType {
     y: u16,
 }
 ```
-In the first case the serialized object of `CompressionType` enum will only contain 4-byte header with the discriminant value set to the selected variant (`0 = GZip`, `1 = Bzip2`). In the latter case apart from the header the serialized data will contain additional bytes according to which variant was selected: 
+In the first case, the serialized object of `CompressionType` enum will only contain a 4-byte header with the discriminant value set to the selected variant (`0 = GZip`, `1 = Bzip2`). In the latter case apart from the header the serialized data will contain additional bytes according to which variant was selected: 
 * `Variant1`: 8 bytes
 * `Variant2`: 6 bytes (the sum of `x` and `y` fields of `SomeType` struct)
 
-A special care need to be taken when deserializing such enum as according to the selected variant number of following data bytes may be different.
+Special care needs to be taken when deserializing such enum as according to the selected variant number of following data bytes may be different.
+
+### PushMessage
+It is sent by nodes who want to share information with others. Nodes gather data from their `crds` and send push messages to their peers periodically.
 
+Nodes receiving the messages check them for duplication, insert them into their `crds` in case the received value doesn't exist or is updated and transmit them further to their peers.
+
+| Data | Type | Size | Description |
+|------|:----:|:----:|-------------|
+| `Pubkey` | `[u8; 32]` | 32 | a public key of the origin |
+| `CrdsValuesList` | [`[CrdsValue]`](#data-shared-between-nodes) | 8+ | a list of values to share  |
+
+<details>
+  <summary>Solana client Rust implementation</summary>
+
+```rust
+enum Protocol {
+    //...
+    PushMessage(Pubkey, Vec<CrdsValue>),
+    //...
+}
+```
+
+</details>
 
 ### PullRequest
-A node sends it to ask the cluster for new information. It contains a bloom filter with things the node already has. Nodes receiving pull requests gather all new values from their `crds`, filter them using provided filters and send `PullResponse` to the origin of the request.
+A node sends a pull request to ask the cluster for new information. It creates a list of bloom filters for its `crds` values and sends different bloom filters to different peers. The recipients of pull requests check what info the sender is missing using the received bloom filter and then construct a [PullResponse](#pullresponse) packed with missing `crds` values data for the pull request origin.
```

Comment:
nit: could just refer to 
> missing `crds` values data for ...

as: missing `CrdsValue`s for ... (or something like that)